core/src/main/scala/caliban/rendering/DocumentRenderer.scala

```diff
@@ -322,6 +335,64 @@ object DocumentRenderer extends Renderer[Document] {
         }
     }
 
+  private trait SchemaOps {
+    def unsafeRenderImpl(
+      theType: String,
+      description: Option[String],
+      directives: List[Directive],
+      query: Option[String],
+      mutation: Option[String],
+      subscription: Option[String],
+      indent: Option[Int],
+      writer: StringBuilder
+    ) = {
+      def renderOp(name: String, op: Option[String]): Unit =
+        op.foreach { o =>
+          pad(increment(indent), writer)
+          writer append name
+          writer append ':'
+          spaceOrEmpty(indent, writer)
+          writer append o
+          newlineOrEmpty(indent, writer)
+        }
+      newlineOrEmpty(indent, writer)
+      descriptionRenderer.unsafeRender(description, indent, writer)
+      writer append theType
+      spaceOrEmpty(indent, writer)
+      directivesRenderer.unsafeRender(directives, indent, writer)
+      val hasTypes                                         = query.nonEmpty || mutation.nonEmpty || subscription.nonEmpty
+      if (hasTypes) {
+        writer append '{'
+        newlineOrEmpty(indent, writer)
+        renderOp("query", query)
+        renderOp("mutation", mutation)
+        renderOp("subscription", subscription)
+        writer append '}'
+      }
+      newlineOrEmpty(indent, writer)
+    }
+  }
+  private lazy val schemaExtensionRenderer: Renderer[TypeSystemExtension.SchemaExtension] =
+    new Renderer[TypeSystemExtension.SchemaExtension] with SchemaOps {
+      override protected[caliban] def unsafeRender(
+        extension: TypeSystemExtension.SchemaExtension,
+        indent: Option[Int],
+        write: StringBuilder
+      ): Unit = {
+        import extension._
+        unsafeRenderImpl("extend schema", None, directives, query, mutation, subscription, indent, write)
+      }
+    }
+  // TODO: The implementation of schemaRenderer is wrong. Two schema definations is invalid. The current implementation
```

Comment:
@paulpdaniels could you check that?

Reply:
Do we want to keep the todo and address it in a follow up then?